async-nats/tests/client_tests.rs

```diff
@@ -618,6 +618,92 @@ mod client {
         rx.recv().await;
     }
 
+    #[tokio::test]
+    async fn reconnect_delay_callback_custom() {
+        let server = nats_server::run_basic_server();
+
+        let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
+
+        let _ = ConnectOptions::new()
+            .retry_on_initial_connect()
+            .reconnect_delay_callback(move |attempts| {
+                let tx = tx.clone();
+
+                let duration =
+                    std::time::Duration::from_millis(std::cmp::min((attempts * 10) as u64, 100));
+                // report back the number of attempts
+                tx.send((attempts, duration)).unwrap();
+
+                duration
+            })
+            .connect(server.client_url())
+            .await
+            .unwrap();
+
+        drop(server);
+
+        let (attempt, duration) = rx.recv().await.unwrap();
```

Comment:
Those are to small Durations to be stable on test runs, especially on CI.

I would work on at least 250ms+ durations.

Set a custom callback that on first attempt tries immediately, then on second after 500ms, then on third - after 1s.
That should be both clear to read and understand what the test is doing and stable.